File: WebShopAppBackend/webshop/src/main/java/org/vs/dto/ProizvodKarticaOdgovor.java
package org.vs.dto;

import jakarta.validation.constraints.Digits;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Positive;
import jakarta.validation.constraints.Size;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class ProizvodKarticaOdgovor {

	@NotBlank
	private Integer id;

	@NotBlank
	@Size(min = 3, max = 30)
	private String kratakNaslov;

	@Positive
	@Digits(integer = 10, fraction = 2)
	@NotNull
	private Double cijena;

	@NotBlank
	private String nazivKategorije;

	@NotBlank
	private String posterUrl;

}

File: WebShopAppFrontend/web-shop-app/src/app/model/proizvodKarticaOdgovor.ts
export class ProizvodKarticaOdgovor {

  id: number;
  kratakNaslov: string;
  cijena: number;
  nazivKategorije: string;
  posterUrl: string;

  constructor(
    id: number,
    kratakNaslov: string,
    cijena: number,
    nazivKategorije: string,
    posterUrl: string
  ) {
    this.id = id;
    this.kratakNaslov = kratakNaslov;
    this.cijena = cijena;
    this.nazivKategorije = nazivKategorije;
    this.posterUrl = posterUrl;
  }

}
